### ISSUE TITLE
Wrong parsing of cell value

### ISSUE THREAD

**Issue by AlbertoBonfiglio** [justkawal/excel#111]
Hi there,
maybe I am doing something wrong but when I read a cell formatted as custom string with value 20997.8364 (so it shows in excel as $20,998)  it actually gets read as:
CellType.String and _value: "1957-06-26T20:04:24.959" 

The cell object shows as : Data (Data(1957-06-26T20:04:24.959, 1, 13, null, Greencross - Vet KPI))

Any help is appreciated

**Comment by hrwn** [comment]
I have same problem, I change format cells in excel file to text and solve it

**Comment by AlbertoBonfiglio** [comment]
Maybe the problem is that by default it converts to string.  It would be interesting to see how it comes in to potentially fix it.

**Comment by SlickSlime** [comment]
But why would the value be in `DateTime` format?

**Comment by SlickSlime** [comment]
#119  Similar to this where int coming out as String (in DateTime format)

**Comment by niklasbartsch** [comment]
I have the same problem and its converting my number 000700 into the DateTime String "1901-11-30 00:00:00.000"

I tried some things and it seems like it is always taking the days since the 30.12.1899 😅

``` dart

        final dateTime = DateTime.tryParse(number);
        print(dateTime.toString());  // 1901-11-30 00:00:00.000
        final differences =
        dateTime!.difference(DateTime(1899, 12, 30)).inDays;
        print('differences: $differences') // 700
```

I would guess it is taking the numbers after the dot for the minutes since this date

Just a quick workaround for now.

**Comment by AlexanderArendar** [comment]
Problem is actual and blocking usage of the package for a very basic case: reading an xlsx file with numeric floating point values.

**Comment by AlexanderArendar** [comment]
Currently we can use following workaround elaborated from the comment of @niklasbartsch :
      final row = sheet.rows[i];
      final rawValue = row.elementAt(columnIndex)!.value as String;
      final date = DateTime.parse(rawValue);
      final diffDays = date.difference(DateTime(1899, 12, 30)).inDays;
      final diffMillis = date.difference(DateTime(1899, 12, 30)).inMilliseconds;
      final partDelta = diffDays * 24 * 3600 * 1000;
      final remainder = diffMillis - partDelta;
      final millisInDay = 24 * 3600 * 1000;
      double fractionalPart = remainder / millisInDay;
      result = diffDays + fractionalPart;

**Comment by DamianBender** [comment]
Hey,
I still have this issue with version 2.0.1.
Here is the output that I get from printing the sheet:
![printofsheet](https://user-images.githubusercontent.com/57224031/222919141-4e006083-56ed-432a-a8c2-9d42bdba10c0.png)
Here is the sheet:
[test.xlsx](https://github.com/justkawal/excel/files/10889207/test.xlsx)

**Comment by FauconSpartiate** [comment]
@justkawal

**Comment by DamianBender** [comment]
Is there any new info about this problem?

**Comment by FauconSpartiate** [comment]
@DamianBender 
Nope, someone motivated enough troubleshoot this is welcome, solving it shouldn't be that complicated...

**Comment by justkawal** [comment]
I'll look into it....

**Comment by DamianBender** [comment]
Hello guys,
it's already been a few months from the last post. 
Can you please give me a short response on that?

**Comment by jv-ribeiros** [comment]
Is there any new info about this?

**Comment by FauconSpartiate** [comment]
Fixed in #279
